fix(infra): lint_diff uses incorrect relative paths in multiple package Makefiles

### Checked other resources

- [x] This is a bug, not a usage question.
- [x] I added a clear and descriptive title that summarizes this issue.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [x] I am sure that this is a bug in LangChain rather than my code.
- [x] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).
- [x] This is not related to the langchain-community package.
- [x] I posted a self-contained, minimal, reproducible example. A maintainer can copy it and run it AS IS.

### Package (Required)

- [x] langchain
- [ ] langchain-openai
- [ ] langchain-anthropic
- [ ] langchain-classic
- [ ] langchain-core
- [ ] langchain-model-profiles
- [ ] langchain-tests
- [x] langchain-text-splitters
- [ ] langchain-chroma
- [ ] langchain-deepseek
- [ ] langchain-exa
- [ ] langchain-fireworks
- [ ] langchain-groq
- [ ] langchain-huggingface
- [x] langchain-mistralai
- [ ] langchain-nomic
- [ ] langchain-ollama
- [ ] langchain-openrouter
- [ ] langchain-perplexity
- [ ] langchain-qdrant
- [ ] langchain-xai
- [ ] Other / not sure / general

### Related Issues / PRs

_No response_

### Reproduction Steps / Example Code (Python)

```python
from pathlib import Path

cases = {
"libs/langchain_v1/Makefile": "git diff --relative=libs/langchain_v1",
"libs/text-splitters/Makefile": "git diff --relative=libs/text-splitters",
"libs/partners/mistralai/Makefile": "git diff --relative=libs/partners/mistralai",
}

for file_path, expected in cases.items():
text = Path(file_path).read_text(encoding="utf-8")
lint_diff_line = next(
line for line in text.splitlines()
if line.startswith("lint_diff format_diff:")
)
print(file_path, "->", lint_diff_line)
assert expected in lint_diff_line, (
f"{file_path} has incorrect --relative path. "
f"Expected to include: {expected}"
)
```

### Error Message and Stack Trace (if applicable)

```shell
AssertionError: libs/langchain_v1/Makefile has incorrect --relative path. Expected to include: git diff --relative=libs/langchain_v1
```

### Description

On current master, lint_diff/format_diff in multiple package Makefiles use incorrect git diff --relative paths.

Expected behavior:
Each Makefile should compute changed files relative to its own package directory.

Actual behavior:
Some Makefiles point to other package paths or a typo path, so changed-file linting can target incorrect files or skip intended package-local changes.

Impact:
Contributor local validation becomes unreliable and issues may only appear later in CI.

Scope:
Infra-only bugfix; no public API/runtime behavior changes.

### System Info

System Information
------------------
> OS:  Linux
> OS Version:  #19~24.04.2-Ubuntu SMP PREEMPT_DYNAMIC Fri Mar  6 23:08:46 UTC 2
> Python Version:  3.12.3 (main, Mar  3 2026, 12:15:18) [GCC 13.3.0]

Package Information
-------------------

Optional packages not installed
-------------------------------
> langsmith
> deepagents
> deepagents-cli

--- a/libs/langchain_v1/Makefile
+++ b/libs/langchain_v1/Makefile
@@ -81,7 +81,7 @@ check_version:
 PYTHON_FILES=.
 MYPY_CACHE=.mypy_cache
 lint format: PYTHON_FILES=.
-lint_diff format_diff: PYTHON_FILES=$(shell git diff --relative=libs/langchain --name-only --diff-filter=d master | grep -E '\.py$$|\.ipynb$$')
+lint_diff format_diff: PYTHON_FILES=$(shell git diff --relative=libs/langchain_v1 --name-only --diff-filter=d master | grep -E '\.py$$|\.ipynb$$')
 lint_package: PYTHON_FILES=langchain
 lint_tests: PYTHON_FILES=tests
 lint_tests: MYPY_CACHE=.mypy_cache_test

--- a/libs/partners/mistralai/Makefile
+++ b/libs/partners/mistralai/Makefile
@@ -32,7 +32,7 @@ integration_test integration_tests:
 PYTHON_FILES=.
 MYPY_CACHE=.mypy_cache
 lint format: PYTHON_FILES=.
-lint_diff format_diff: PYTHON_FILES=$(shell git diff --relative=libs/partners/mistral --name-only --diff-filter=d master | grep -E '\.py$$|\.ipynb$$')
+lint_diff format_diff: PYTHON_FILES=$(shell git diff --relative=libs/partners/mistralai --name-only --diff-filter=d master | grep -E '\.py$$|\.ipynb$$')
 lint_package: PYTHON_FILES=langchain_mistralai
 lint_tests: PYTHON_FILES=tests
 lint_tests: MYPY_CACHE=.mypy_cache_test

--- a/libs/text-splitters/Makefile
+++ b/libs/text-splitters/Makefile
@@ -37,7 +37,7 @@ extended_tests:
 PYTHON_FILES=.
 MYPY_CACHE=.mypy_cache
 lint format: PYTHON_FILES=.
-lint_diff format_diff: PYTHON_FILES=$(shell git diff --relative=libs/core --name-only --diff-filter=d master | grep -E '\.py$$|\.ipynb$$')
+lint_diff format_diff: PYTHON_FILES=$(shell git diff --relative=libs/text-splitters --name-only --diff-filter=d master | grep -E '\.py$$|\.ipynb$$')
 lint_package: PYTHON_FILES=langchain_text_splitters
 lint_tests: PYTHON_FILES=tests/unit_tests
 lint_tests: MYPY_CACHE=.mypy_cache_test

